### PR TITLE
Generate overview of branch statistics, quick hack

### PR DESCRIPTION
**Pull request by smarr**
This is a quick hack to accumulate the an overview of that data collected. From my understanding, these are the interesting bits, but I might be wrong.

More work needed. That's merely a proposal to perhaps make the data a little more digestible.

With this change, the output is accumulated to produce entries like this:
```
at som.interpreter.nodes.dispatch.CachedDispatchNode.executeDispatch(CachedDispatchNode.java:38) [bci: 10]
NONE: 29 IF: 3 ELSE: 5 BOTH: 33
```

### CODE REVIEW

**Comment by axel22** [comment]
Looks good (and useful) to me! I recently pushed this commit in which I use execution counts per each if-branch, instead of just boolean flags:

https://github.com/graalvm/graal-core/commit/c3f892aba6d24a9fe9f2f5b0469e1dcc96ddad03

File: graal/com.oracle.graal.truffle/src/com/oracle/graal/truffle/phases/InstrumentBranchesPhase.java [hunk 1]
```diff
@@ -144,8 +146,43 @@ private static String filterAndEncode(Node ifNode) {
             }
         }
 
+        private static class PointStats {
+            long NONE;
+            long IF;
+            long ELSE;
+            long BOTH;
+
+            public String toString() {
```
Comment (tkrodriguez) [inline graal/com.oracle.graal.truffle/src/com/oracle/graal/truffle/phases/InstrumentBranchesPhase.java]: You need @Override here.

Comment (smarr) [inline graal/com.oracle.graal.truffle/src/com/oracle/graal/truffle/phases/InstrumentBranchesPhase.java]: Yeah, the code also uses lambdas, so not sure this is really desirable in Graal.

Comment (tkrodriguez) [inline graal/com.oracle.graal.truffle/src/com/oracle/graal/truffle/phases/InstrumentBranchesPhase.java]: The lambda's were already there so I wouldn't worry about that.  But it's definitely a case where the number of characters would be the same without lambdas, but it wouldn't be formatted as one line.  Lambdas seem to really lose on the conciseness front.

**Pull request closed without merging**